components/infra-proxy-service/server/util.go

```diff
@@ -0,0 +1,41 @@
+package server
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// SafeStringFromMap returns the value referenced by `key` in `values`. If value is nil,
+// it returns an empty string; otherwise it returns the original string.
```

Comment:
I think this conveys the intent more clearly:
```suggestion
// SafeStringFromMap returns the value referenced by `key` in `values` as a string.
// If not found, it returns an empty string.
```

Reply:
Thanks!